### PR TITLE
feature: add default registry namespace

### PR DESCRIPTION
**Pull request by Ace-Tang**
Signed-off-by: Ace-Tang <aceapril@126.com>

### Ⅰ. Describe what this PR did

add registry namespace in config, it will be used in pull/push/search images.

### CODE REVIEW

**Comment by HusterWan** [comment]
LGTM

File: daemon/mgr/image_utils.go [hunk 1]
```diff
@@ -19,7 +20,7 @@ func (mgr *ImageManager) addRegistry(input string) string {
 	if _, ok := reference.Domain(input); ok {
 		return input
 	}
-	return mgr.DefaultRegistry + input
+	return filepath.Join(mgr.DefaultRegistry, mgr.DefaultNamespace, input)
```
Comment (HusterWan) [inline daemon/mgr/image_utils.go]: That means we only specify the image name after merge this pr, just like `pouch pull centos`?

File: main.go [hunk 2]
```diff
@@ -81,7 +81,8 @@ func setupFlags(cmd *cobra.Command) {
 	flagSet.BoolVar(&cfg.IsLxcfsEnabled, "enable-lxcfs", false, "Enable Lxcfs to make container to isolate /proc")
 	flagSet.StringVar(&cfg.LxcfsBinPath, "lxcfs", "/usr/local/bin/lxcfs", "Specify the path of lxcfs binary")
 	flagSet.StringVar(&cfg.LxcfsHome, "lxcfs-home", "/var/lib/lxc/lxcfs", "Specify the mount dir of lxcfs")
-	flagSet.StringVar(&cfg.DefaultRegistry, "default-registry", "registry.hub.docker.com/library/", "Default Image Registry")
+	flagSet.StringVar(&cfg.DefaultRegistry, "default-registry", "registry.hub.docker.com", "Default Image Registry")
```
Comment (HusterWan) [inline main.go]: I have not got your point yet, why we need split this params into two part? can you explain it, thanks a lot

File: daemon/mgr/image_utils.go (hunk 1 above)
Comment (Ace-Tang) [reply daemon/mgr/image_utils.go]: no, it just follow original logic, not change it, what I do is to split docker.io/library into docker.io + library.

File: main.go (hunk 2 above)
Comment (Ace-Tang) [reply main.go]: for fix https://github.com/alibaba/pouch/issues/892, I will add this issue into my commit.

**Pull request merged**